graylog2-web-interface/src/components/inputs/InputSetupWizard/InputSetupWizard.tsx

```diff
@@ -15,30 +15,82 @@
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
 import * as React from 'react';
+import { useEffect, useCallback, useMemo, useState } from 'react';
+import { PluginStore } from 'graylog-web-plugin/plugin';
 
 import { Modal } from 'components/bootstrap';
 import { Wizard } from 'components/common';
-import { INPUT_WIZARD_STEPS } from 'contexts/InputSetupWizardContext';
-import useInputSetupWizard from 'hooks/useInputSetupWizard';
+import { INPUT_WIZARD_STEPS } from 'components/inputs/InputSetupWizard/types';
+import useInputSetupWizard from 'components/inputs/InputSetupWizard/hooks/useInputSetupWizard';
 
-import CategoryStep from './CategoryStep';
+import { SelectCategoryStep, TestInputStep } from './steps';
 
 const InputSetupWizard = () => {
-  const { activeStep, setActiveStep, show, closeWizard } = useInputSetupWizard();
-
-  const steps = [{
-    key: INPUT_WIZARD_STEPS.SELECT_CATEGORY,
-    title: (
-      <>
-        Title
-      </>
-    ),
-    component: (
-      <CategoryStep />
-    ),
-  }];
-
-  if (!show) return null;
+  const { activeStep, setActiveStep, show, closeWizard, wizardData } = useInputSetupWizard();
+  const [orderedSteps, setOrderedSteps] = useState([]);
+  const { category, subcategory } = wizardData;
+  const enterpriseSteps = PluginStore.exports('inputSetupWizard').find((plugin) => (!!plugin.steps))?.steps;
+
+  const steps = useMemo(() => {
```

Comment:
Hmm, I don't agree. The only purpose of this component is to handle the logic for the input wizard. I don't see why we should extract it into a hook, to end up with a wrapper component and a hook that will not be reused. I think in this case it won't help by reducing complexity, maybe it will even increase it due to all the files.

Reply:
I'm not sure about that, but it's okay we can keep it as it is.